src/Servers/Kestrel/Core/src/Internal/Http2/Http2MessageBody.cs

```diff
@@ -63,14 +64,23 @@ public override void AdvanceTo(SequencePosition consumed)
 
         public override void AdvanceTo(SequencePosition consumed, SequencePosition examined)
         {
-            var dataLength = _readResult.Buffer.Slice(_readResult.Buffer.Start, consumed).Length;
+            var dataLength = _readResult.Buffer.Slice(_readResult.Buffer.Start, examined).Length;
+            var newlyExamined = dataLength - _totalExamined;
```

Comment:
This only make sense if _totalExamined is _totalExamined per _readResult. You would need to reset _totalExamined when you update _readResult.

Reply:
You are correct. I think I can fix this via using consumed. I'll update it shortly.